src/test/java/seedu/address/logic/commands/AddBuyersToListingCommandTest.java

```diff
@@ -0,0 +1,325 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.math.BigDecimal;
+import java.nio.file.Path;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Listings;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.ReadOnlyListings;
+import seedu.address.model.ReadOnlyUserPrefs;
+import seedu.address.model.listing.Address;
+import seedu.address.model.listing.Area;
+import seedu.address.model.listing.Listing;
+import seedu.address.model.listing.Price;
+import seedu.address.model.listing.Region;
+import seedu.address.model.person.Buyer;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Seller;
+import seedu.address.testutil.PersonBuilder;
+
+public class AddBuyersToListingCommandTest {
+    private static final Name VALID_LISTING_NAME = new Name("Valid Listing Name");
+    private static final Name INVALID_LISTING_NAME = new Name("Invalid Listing Name");
+    private static final Address VALID_ADDRESS = new Address("Valid Address");
+    private static final Price VALID_PRICE = new Price("500k",
+            new BigDecimal(5000000));
+    private static final Area VALID_AREA = new Area(999);
+    private static final Region VALID_REGION = Region.WEST;
+    private static final Person VALID_SELLER = new PersonBuilder().buildSeller();
+    private static final Set<Person> VALID_BUYERS = Set.of(new PersonBuilder().buildBuyer());
+
+    private static final Listing VALID_LISTING = new Listing(VALID_LISTING_NAME,
+            VALID_ADDRESS,
+            VALID_PRICE,
+            VALID_AREA,
+            VALID_REGION,
+            VALID_SELLER,
+            VALID_BUYERS);
+
+    @Test
+    public void constructor_nullPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddBuyersToListingCommand(VALID_LISTING_NAME,
+                null));
+    }
+
+    /*
+    @Test
+    public void constructor_nullListing_throwsCommandException() throws CommandException {
```

Comment:
This should be tested as well

Reply:
sorry love, im not strong enough to do it